scripts/diff-pkg-changes.py

```diff
@@ -0,0 +1,291 @@
+#!/usr/bin/env python3
+
+# Authors: Dylan Turner
+# Description:
+# - Diff feeds of two NI Linux RT releases from their GitHub branches to update FEED_CHANGELOG.md
+# - Steps:
+#     1. Pull core/ and extra/ feed info from GitHub for branches to compare
+#     2. Pull the Packages files from all four feeds
+#     3. Parse the package
+#     4. Compare the packages to each other
+#     5. Generate report
+#     6. Add report to FEED_CHANGELOG.md
+# - Note: Try to keep as few non-base python packages as possible to keep it portable
+# - Example (finding differences between 8.11 and 9.0):
+#     ./scripts/diff-pkg-changes.py \
+#         nilrt/21.8/sumo 8.11 nilrt/22.5/hardknott 9.0 \
+#         --url http://nickdanger.amer.corp.natinst.com/feeds
+
+from argparse import ArgumentParser
+from requests import get, Response
+from dataclasses import dataclass
+
+# Defines the base URL for feeds
+DEF_FEED_BASE: str = 'http://download.ni.com/ni-linux-rt/feeds'
+
+# Constants used for constructing the url to a file on GitHub containing feed information
+GITHUB_BASE: str = 'https://raw.githubusercontent.com/ni/meta-nilrt'
+GITHUB_CONF_FNAME: str = 'conf/distro/nilrt.conf'
+
+# The start text of the FEED_CHANGELOG.md file
+FEED_CHANGELOG_START: str = \
+    '# Changes to NI Linux RT Package Feeds\n\n' \
+        + '## Description\n\n' \
+        + 'This documents the full set of changes to individual packages ' \
+        + 'in the core/ and extra/ feeds for each version release.\n\n' \
+        + 'The changes are too long to keep in the [Changelog](/CHANGELOG.md) itself, ' \
+        + 'so they are kept here\n\n'
+
+def main() -> None:
+    # Get user input
+    settings: 'Settings' = Settings()
```

Comment:
I think you primarily broke out `Settings` into its own class in order to keep the code neat, but I might suggest a slightly different architecture that will keep the code more reusable.

```python3
class CLIParser(ArgumentParser):

    def __init__(self):
        super().__init__(self)
        self.add_argument(...
        ...

def main(args) -> int:
    old_branch: str = args.old_branch
    ...
    return 0

if __name__ == "__main__":
    parser = CLIParser()
    args = parser.parse_args()
    sys.exit(main(args))
```

That way, if someone wants to reuse this module from python, they can source the module and use the `CLIParser` to parse their own argument values (or populate the `args` namespace manually), and then call the `main()` method - all from within python.